pycargoebuild/__main__.py

```diff
@@ -71,13 +72,30 @@ def main(prog_name: str, *argv: str) -> int:
     load_license_mapping(args.license_mapping)
     args.license_mapping.close()
 
+    def iterate_parents(directory: Path) -> typing.Generator[Path, None, None]:
+        root = directory.absolute().root
+        yield directory
+        while not directory.samefile(root):
+            directory /= ".."
+            yield directory
+
+    def get_cargo_lock_file(directory: Path) -> io.BufferedReader:
+        for directory in iterate_parents(directory):
+            try:
+                return open(directory / "Cargo.lock", "rb")
+            except FileNotFoundError as e:
+                if 'err' not in locals():
+                    err = e
```

Comment:
I didn't like the idea of heaving one extra `try-except` out of the loop just to catch the frist exception. This `if` has the same effect.